src/Tooltip/Tooltip.js

```diff
@@ -4,91 +4,183 @@ import ReactDOM from 'react-dom';
 import Util from '../Util/Util';
 
+const ARROW_SIZE = 7;
+
 class Tooltip extends Util.mixin(BindMixin) {
   get methodsToBind() {
     return [
+      'dismissTooltip',
       'getIdealLocation',
       'handleMouseEnter',
-      'handleMouseLeave'
+      'handleMouseLeave',
+      'handleTooltipMouseEnter',
+      'handleTooltipMouseLeave'
     ];
   }
 
   constructor() {
-    super();
-
+    super(...arguments);
+    this.container = null;
     this.state = {isOpen: false};
   }
 
-  getAnchor(anchor, position, renderedPosition, viewportWidth, viewportHeight) {
-    // Change the anchor if the tooltip will be rendered off the screen.
-    if (position === 'right' || position === 'left') {
-      if (renderedPosition.top < 0) {
-        anchor = 'start';
-      } else if (renderedPosition.bottom > viewportHeight) {
-        anchor = 'end';
-      }
-    } else if (position === 'top' || position === 'bottom') {
-      if (renderedPosition.right > viewportWidth) {
-        anchor = 'end';
-      } else if (renderedPosition.left < 0) {
-        anchor = 'start';
+  componentWillUnmount() {
+    this.removeScrollListener();
+  }
+
+  handleMouseEnter(currentAnchor, currentPosition) {
+    let {anchor, position, coordinates} = this.getIdealLocation(currentAnchor,
+      currentPosition);
+    this.setState({anchor, isOpen: true, position, coordinates});
+    this.addScrollListener();
+  }
+
+  handleMouseLeave() {
+    this.dismissTooltip();
+  }
+
+  handleTooltipMouseEnter() {
+    if (this.props.interactive) {
+      this.setState({isOpen: true});
+      this.addScrollListener();
+    }
+  }
+
+  handleTooltipMouseLeave() {
+    this.dismissTooltip();
+  }
+
+  addScrollListener() {
+    if (!this.container) {
+      if (typeof this.props.scrollContainer === 'string') {
+        this.container = DOMUtil.closest(ReactDOM.findDOMNode(this),
+          this.props.scrollContainer) || window;
+      } else {
+        this.container = this.props.scrollContainer;
       }
     }
 
-    return anchor;
+    this.container.addEventListener('scroll', this.dismissTooltip);
+  }
+
+  dismissTooltip() {
+    if (this.state.isOpen) {
+      this.setState({isOpen: false});
+      this.removeScrollListener();
+    }
+  }
+
+  getAnchor(isVertical, anchor, clearance, tooltipWidth, tooltipHeight) {
+    // Calculate the ideal anchor.
+    if (isVertical) {
+      return this.transformAnchor(anchor, clearance.left, clearance.right,
+        tooltipWidth, clearance.boundingRect.width);
+    }
+
+    return this.transformAnchor(anchor, clearance.top, clearance.bottom,
+      tooltipHeight, clearance.boundingRect.height);
   }
 
-  getPosition(position, renderedPosition, viewportWidth, viewportHeight) {
+  getCoordinates(position, clearance, tooltipWidth, tooltipHeight) {
+    // Calculate the coordinates of the tooltip content.
+    if (position === 'top') {
+      return {
+        left: clearance.boundingRect.left + clearance.boundingRect.width / 2,
```

Comment:
These are actually different values... kind of. The `top`, `right`, `bottom`, and `left` properties directly on the `clearance` object reflect the number of pixels between the edge of the viewport and the edge of the element. The `boundingRect` object contains the position of the element on the screen. So while sometimes these values are the same (`top` and `left` will be the same for both values), their intentions are different.